exla/README.md

```diff
@@ -48,6 +48,22 @@ EXLA relies on the [XLA](https://github.com/elixir-nx/xla) package to provide th
 
 For cross-compilation, you need to [set your `XLA_TARGET_PLATFORM` variable](https://github.com/elixir-nx/xla#xla_target_platform) to the correct target platform value (i.e. `aarch64-linux-gnu` for the Raspberry Pi 4).
 
+## Troubleshooting
+
+EXLA uses NIFs (C-interface code called from Elixir) for part of its functionality.
+If for any reason these fail to compile or load, troubleshooting can be tricky.
+
+We recommend following the steps below:
+
+  1. If the error appeared after upgrading EXLA, ensure that you have the proper dependency versions given by [XLA](https://github.com/elixir-nx/xla). Afterwards, compile with `mix compile --clean-libexla-cache --force` to clean up all cached files.
```

Comment:
`mix compile` is not going to work when compiled as a dependency. We would need to introduce a separate mix compiler or a separate task.

Reply:
It won’t work in the sense that it won’t invoke the code you added in the mix.exs file. `mix compile` is only about the local project. Maybe an env var with `mix deps.compile exla` will work.

Reply:
@josevalim it may works as `mix deps.clean exla --build --clean-libexla-cache`?

Reply:
err `mix deps.compile exla --force --clean-libexla-cache`

Reply:
But yeah, env var sounds better either way.

Reply:
I am not sure if we pass options forward in all of these cases. Although we should probably make `mix deps.clean` for dependencies extensible, so we actually invoke `clean` in the relevant dep.